CPVMigrate/Program.cs

```diff
@@ -24,45 +29,86 @@ public static void ProcessProjects(string rootFilePath)
+            Console.WriteLine($"Processed projects are: {string.Join(Environment.NewLine, projects)}...");
+            Console.ReadKey();
         }
 
         public static IEnumerable<NuGetPackage> CollectPackages(string csprojPath)
         {
             var packages = new List<NuGetPackage>();
-            XDocument projectXml;
             using (var stream = File.Open(csprojPath, FileMode.Open, FileAccess.Read, FileShare.Read))
             {
-                projectXml = XDocument.Load(stream, LoadOptions.SetLineInfo);
-                var packagesElements = projectXml.Root.XPathSelectElements("//PackageReference");
-                foreach (var element in packagesElements)
+                var projectXml = XDocument.Load(stream, LoadOptions.SetLineInfo);
+                var packagesElements = projectXml.Root.XPathSelectElements("//PackageReference").ToList();
+
+                if (packagesElements.Count.Equals(0))
+                {
+                    /* https://stackoverflow.com/a/84018408 */
+                    var namespaceManager = new XmlNamespaceManager(new NameTable());
+                    namespaceManager.AddNamespace("prefix", "http://schemas.microsoft.com/developer/msbuild/2003");
+                    packagesElements = projectXml.Root.XPathSelectElements("//prefix:PackageReference", namespaceManager).ToList();
+                }
+
+                // This is to make the tool idempotent. To avoid exception when we run the tool again on same project again.
+                var packagesWithVersion = packagesElements.Where(e => e.Attribute("Version") != null || e.Value != null).ToList();
+
+                foreach (var element in packagesWithVersion)
                 {
                     var p = new NuGetPackage()
                     {
                         Name = element.Attribute("Include").Value,
-                        Version = element.Attribute("Version").Value
+
+                        // element.Value is useful in case of Type-2 mentioned below. 
+                        Version = element.Attribute("Version")?.Value ?? element.Value
                     };
                     packages.Add(p);
                     Console.WriteLine($"   Processing {p.Name}");
                 }
+
+                Console.WriteLine($"   Found {packagesElements.Count} Packages in total.");
+                Console.WriteLine($"   Found {packagesWithVersion.Count} Packages with version attribute.");
                 Console.WriteLine($"   --------------------");
-                Console.WriteLine($"   Found {packagesElements.Count()} Packages Total.");
             }
             return packages;
         }
 
         public static void RemovePackageVersions(string csprojPath)
         {
-            File.WriteAllText(csprojPath, Regex.Replace(File.ReadAllText(csprojPath), "(PackageReference\\sInclude=\"[a-zA-Z0-9\\.]+\")(\\sVersion=\"[a-zA-Z0-9\\.-]+\")", "$1"), new UTF8Encoding(true));
```

Comment:
There were some references like below which was not covered. 
`<PackageReference Include="Node.js" GeneratePathProperty="true" Version="8.10.0" />`